pandas/core/frame.py

```diff
@@ -6998,17 +7000,16 @@ def append(self, other, ignore_index=False, verify_integrity=False, sort=None):
             other = other._convert(datetime=True, timedelta=True)
             if not self.columns.equals(combined_columns):
                 self = self.reindex(columns=combined_columns)
-        elif isinstance(other, list) and not isinstance(other[0], DataFrame):
+        elif isinstance(other, list) and other and not isinstance(other[0], DataFrame):
             other = DataFrame(other)
             if (self.columns.get_indexer(other.columns) >= 0).all():
                 other = other.reindex(columns=self.columns)
 
-        from pandas.core.reshape.concat import concat
-
-        if isinstance(other, (list, tuple)):
```

Comment:
Support for `tuple` didn't work as the line below would raise a `TypeError`. As tuple support isn't documented, I decided to remove it here so that it won't confuse other readers as well.